tests/func/test_add.py

```diff
@@ -245,9 +244,15 @@ def test_dir(self):
         self.assertEqual(ret, 0)
 
 
-class TestShouldUpdateStateEntryForFileAfterAdd(TestDvc):
+class TestDvcWithMocker(TestDvc):
+    @pytest.fixture(autouse=True)
+    def use_mocker(self, mocker):
+        self.mocker = mocker
+
+
+class TestShouldUpdateStateEntryForFileAfterAdd(TestDvcWithMocker):
     def test(self):
-        file_md5_counter = spy(dvc.remote.local.file_md5)
+        file_md5_counter = self.mocker.spy(dvc.remote.local, "file_md5")
         with patch.object(dvc.remote.local, "file_md5", file_md5_counter):
```

Comment:
With mocker spy, there is no need to patch, it is automatically done by mocker, so with statement can be removed

Reply:
Great to know, thanks!